source/slang/slang-emit-spirv.cpp

```diff
@@ -6330,28 +6356,6 @@ SlangResult emitSPIRVFromIR(
         context.ensureInst(irEntryPoint);
     }
 
-    // Declare integral input builtins as Flat if necessary.
-    for (auto globalInst : context.m_irModule->getGlobalInsts())
```

Comment:
I am trying to understand why the issue was happening with the previous code.
It looks like it shouldn't have emitted the duplicated decoration.
Did this for-loop end up iterating a same instance more than once?
Can you share some information of what the problem was?

Reply:
The duplication is introduced because the code here didn't check if `globalInst` already has a `InterpolationModeDecoration`. The only reason to have the loop here to make sure the spirv builtin variables we added during spirv-emit has proper `flat` decorations, the logic is not required for all other user defined global variables, which is already handled by glsl-legalization pass.

I cleaned this up by moving the logic here to `getBuiltinGlobalVar` instead, so we don't double decorate ordianry user-defined global vars.